Automattic/wp-calypso
# Preview limited Global Styles by default

Fixes https://github.com/Automattic/dotcom-forge/issues/2267

## Proposed Changes

- Custom styles are previewed by default, so there is less confusion around the site-editor and the front-end not matching up.
- Both the site editor and the post editor display now a global notice when Global Styles are limited. Previous sidebar notices in the site editor have been removed. This notice has the following actions:
  - Upgrade now (primary): Visible in both editors – It opens the Plans page in a new tab.
  - Remove custom styles (link): Visible in the site editor only – It resets the styles to defaults (it’s easy and safe to do so in the editor). Once we have a support page explaining how to remove custom styles, we can show it in the post editor too.
- Add a new notice to the view canvas.
- Updates the E2E test to use the view canvas mode since that's the only place where we have a fragile CSS selector.


**Frontend**

Before | After
--- | ---
<img width="1279" alt="Screenshot 2023-05-18 at 11 03 58" src="https://github.com/Automattic/wp-calypso/assets/1233880/5a11ae7f-6ffc-4b16-a9dd-6d86cb755396"> | <img width="1278" alt="Screenshot 2023-05-18 at 11 06 33" src="https://github.com/Automattic/wp-calypso/assets/1233880/aecf98f5-f0fa-4500-9c1d-5be68f641583">

**Site editor (view canvas)**

Before | After
--- | ---
<img width="1277" alt="Screenshot 2023-05-18 at 11 03 15" src="https://github.com/Automattic/wp-calypso/assets/1233880/2a9c0e5e-ce55-49b3-98cb-edd28de9ba50"> | <img width="1279" alt="Screenshot 2023-05-18 at 11 05 03" src="https://github.com/Automattic/wp-calypso/assets/1233880/36c8917f-6e36-4c06-ab9d-cb5941f3a2b8">


**Site editor (edit canvas)**

Before | After
---  | ---
<img width="1279" alt="Screenshot 2023-05-18 at 11 03 27" src="https://github.com/Automattic/wp-calypso/assets/1233880/3242bece-9bf8-41c9-b8a0-df55ee80bab1"> | <img width="1279" alt="Screenshot 2023-05-18 at 11 09 01" src="https://github.com/Automattic/wp-calypso/assets/1233880/66ce418c-8fb5-4072-bcde-a9aabb00119d">

**Post editor**

Before | After
--- | ---
<img width="1277" alt="Screenshot 2023-05-18 at 11 03 47" src="https://github.com/Automattic/wp-calypso/assets/1233880/76e1b136-4deb-4759-b2e9-d842f47e0973"> | <img width="1279" alt="Screenshot 2023-05-18 at 11 06 20" src="https://github.com/Automattic/wp-calypso/assets/1233880/16af1bb8-2a8e-4688-bf73-d66d14b12ee0">



## Testing Instructions

- Apply these changes to your sandbox.
- Sandbox a free site that uses Global Styles.
- Open the site editor.
- Make sure there is a global notice at the top of the editor.
- Make sure the "Upgrade now" action opens the Plans page.
- Make sure the "Remove custom styles" action resets the styles to defaults.
- Make sure the notice goes aways when there aren't custom styles.
- Open the post editor.
- Make sure the custom styles are visible.
- Make sure there is a global notice at the top of the editor.
- Make sure the "Upgrade now" action opens the Plans page.
- Make sure the "Preview your site" opens the frontend without custom styles previewed.
- Visit the frontend.
- Make sure custom styles are previewed by default.
- Use the Calypso live link below.
- Go to `/setup/newsletter`.
- Pick a "premium" color.
- As soon as the site is created, sandbox it.
- Get to the Launchpad's last step.
- Ensure the preview is using the "premium" color.
- Follow E2E tests setup instructions from https://github.com/Automattic/wp-calypso/tree/trunk/test/e2e
- Sandbox the primary site of the `simpleSiteFreePlanUser` user.
- Run the E2E test:
  - Desktop: `yarn workspace wp-e2e-tests test -- specs/fse/fse__limited-global-styles.ts`
  - Mobile: `yarn workspace wp-e2e-tests test:mobile -- specs/fse/fse__limited-global-styles.ts`
